app/tests/conftest.py

```diff
@@ -0,0 +1,23 @@
+from typing import Any
+
+from pytest import fixture
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
+
+
+@fixture
+def user_factory(db):
+
+    def create_user_with_permissions(
+        username: str, password: str, permissions: list[tuple[str, str, str]]
+    ) -> Any:
+        user = get_user_model().objects.create_user(username=username, password=password)
+        for app_label, model, codename in permissions:
+            content_type = ContentType.objects.get(app_label=app_label, model=model)
```

Comment:
Maybe add a little comment here why this is necessary.

Reply:
Done